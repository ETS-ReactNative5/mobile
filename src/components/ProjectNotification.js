import React, { Component } from 'react';
import {
  Switch,
  View
} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet'
import PropTypes from 'prop-types';
import theme from '../theme'
import StyledText from './StyledText'
import { connect } from 'react-redux'
import { setState, syncNotificationStore, updateInterestSubscription } from '../actions/index'

const mapStateToProps = (state, ownProps) => ({
  notification: state.main.notifications[ownProps.id]
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  updateProjectNotification(checked) {
    // Implement Firebase Project Notification change
  },
})

export class ProjectNotification extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Switch
          value={this.props.notification}
          style={styles.switchComponent}
          onTintColor={theme.$headerColor}
          onValueChange={(checked) => this.props.updateProjectNotification(checked)}
        />

        <StyledText text={this.props.name} />
      </View>
    );
  }
}

const styles = EStyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 8,
    alignItems: 'center'
  },
  switchComponent: {
    marginRight: 10
  },
});

ProjectNotification.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  notification: PropTypes.bool.isRequired,
  updateProjectNotification: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(ProjectNotification)
