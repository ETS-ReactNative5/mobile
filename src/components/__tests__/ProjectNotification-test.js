import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ProjectNotification } from '../ProjectNotification'

jest.mock('Switch', () => 'Switch');

it('renders checked', () => {
  const tree = renderer.create(
    <ProjectNotification id={'1'} name={'test'} notification={true} updateProjectNotification={jest.fn} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders not checked', () => {
  const tree = renderer.create(
    <ProjectNotification id={'1'} name={'test'} notification={false} updateProjectNotification={jest.fn} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
