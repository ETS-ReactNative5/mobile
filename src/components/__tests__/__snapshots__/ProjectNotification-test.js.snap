exports[`test renders checked 1`] = `
<View
  style={undefined}>
  <Switch
    onTintColor="rgba(0, 151, 157, 1)"
    onValueChange={[Function]}
    style={undefined}
    value={true} />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
      ]
    }>
    test
  </Text>
</View>
`;

exports[`test renders not checked 1`] = `
<View
  style={undefined}>
  <Switch
    onTintColor="rgba(0, 151, 157, 1)"
    onValueChange={[Function]}
    style={undefined}
    value={false} />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
      ]
    }>
    test
  </Text>
</View>
`;
